tools/targets/__init__.py

```diff
@@ -707,6 +707,46 @@ def binary_hook(t_self, resources, elf, binf):
         )
         lpc55s69_complete(t_self, binf, secure_bin)
 
+class M2351Code:
+    """M2351 Hooks"""
+    @staticmethod
+    def merge_secure(t_self, resources, ns_elf, ns_hex):
+        t_self.notify.info("Merging non-secure image with secure image")
+        configured_secure_image_filename = t_self.target.secure_image_filename
+        t_self.notify.info("Non-secure elf image %s" % ns_elf)
+        t_self.notify.info("Non-secure hex image %s" % ns_hex)
+        t_self.notify.info("Finding secure image %s" % configured_secure_image_filename)
+        s_hex = find_secure_image(
+            t_self.notify,
+            resources,
+            ns_hex,
+            configured_secure_image_filename,
+            FileType.HEX
+        )
+        t_self.notify.info("Found secure image %s" % s_hex)
+        
+        _, ext = os.path.splitext(s_hex)
+        assert ext == ".hex", "Secure image %s must be in Intel HEX format" % s_hex
+        assert os.path.isfile(s_hex), "Secure image %s must be regular file" % s_hex
+        
+        _, ext = os.path.splitext(ns_hex)
+        assert ext == ".hex", "Non-secure image %s must be in Intel HEX format" % s_hex
+        assert os.path.isfile(ns_hex), "Non-secure image %s must be regular file" % s_hex
+
+        # Keep original non-secure before merge with secure
+        ns_nosecure_hex = _ + "_no-secure-merge" + ext
```

Comment:
Please don't use `_` as variable name, convention says this can be used when a return value is not required but then to actually use it is wrong.

Reply:
Fixed